IBF-Typhoon-model/src/typhoonmodel/utility_fun/Rainfall_data.py

```diff
@@ -62,6 +70,34 @@ def download_rainfall_nomads(Input_folder, path, Alternative_data_point):
             logger.info(f'Writing to file {filepath}')
             ds = ds.median(dim='number') #store only the median of the ensemble members
             ds.to_netcdf(filepath)
+        #zonal stats to calculate rainfall per manucipality 
+        rain_6h=rasterio.open(filepath) 
+        band_indexes = rain_6h.indexes
+        transform = rain_6h.transform
+        all_band_summaries = []
+        for b in band_indexes:
+            array = rain_6h.read(b)
+            band_summary = zonal_stats(
+                admin,
+                array,
+                prefix=f"band{b}_",
+                stats="mean",
+                nodata=no_data_value,
+                all_touched=True,
+                affine=transform,
+            )
+            all_band_summaries.append(band_summary)
+        # Flip dimensions
+        shape_summaries = list(zip(*all_band_summaries))
+        # each list entry now reflects a municipalities, and consists of a dictionary with the rainfall in mm / 6h for each time frame
+        final = [{k: v for d in s for k, v in d.items()} for s in shape_summaries]
+        # Obtain list with maximum 6h rainfall
+        maximum_6h = [max(x.values()) for x in final]
```

Comment:
This function is getting a bit long. I think it would be cleaner if you refactored all the steps needed to obtain `maximum_6h` into their own function.